include/seqan/seq_io/sequence_file.h

```diff
@@ -68,40 +69,45 @@ typedef FormattedFile<Fastq, Input>     SeqFileIn;
 
 typedef FormattedFile<Fastq, Output>    SeqFileOut;
 
-// --------------------------------------------------------------------------
-// Tag AutoSeqFormat
-// --------------------------------------------------------------------------
-// if TagSelector is set to -1, the file format is auto-detected
-
-/*!
- * @class AutoSeqFormat
- * @extends TagSelector
- * @headerfile <seqan/file.h>
- * @brief Auto-detects and stores a file format.
- *
- * @signature typedef TagList<Fastq, TagList<Fasta, TagList<Raw> > > SeqFormats;
- * @signature typedef TagSelector<SeqFormat> AutoSeqFormat;
- */
 
 typedef
     TagList<Fastq,
     TagList<Fasta,
     TagList<Embl,
     TagList<GenBank,
-    TagList<Raw
-    > > > > >
+    TagList<Raw,
+    TagList<Bam,
+    TagList<Sam
+    > > > > > > >
     SeqInFormats;
 
 typedef
     TagList<Fastq,
     TagList<Fasta,
-    TagList<Raw
-    > > >
+    TagList<Raw,
+    TagList<Bam,
+    TagList<Sam
+    > > > > >
     SeqOutFormats;
 
+
 typedef TagSelector<SeqInFormats>   SeqInFormat;
 typedef TagSelector<SeqOutFormats>  SeqOutFormat;
 
+// --------------------------------------------------------------------------
+// Tag AutoSeqFormat
+// --------------------------------------------------------------------------
+// if TagSelector is set to -1, the file format is auto-detected
+
+/*!
+ * @class AutoSeqFormat
+ * @extends TagSelector
+ * @headerfile <seqan/file.h>
+ * @brief Auto-detects and stores a file format.
+ *
+ * @signature typedef TagList<Fastq, TagList<Fasta, TagList<Raw> > > SeqFormats;
+ * @signature typedef TagSelector<SeqFormat> AutoSeqFormat;
+ */
```

Comment:
hm, try not to mix "general improvements / code refactoring" with regular PRs as it will create problems later on.